src/session_impl.cpp

```diff
@@ -5539,6 +5545,11 @@ namespace {
 			return std::uint16_t(sock->tcp_external_port());
 		}
 
+		if (m_settings.get_int(settings_pack::proxy_type) != settings_pack::none
+			&& m_settings.get_bool(settings_pack::proxy_peer_connections)
+			&& !m_settings.get_bool(settings_pack::proxy_accept_incoming))
+			return 0;
```

Comment:
This check and early exit should only be used if `sock` == `nullptr`. If `sock` is set, we already have this check based on whether the listen socket is using a proxy or not.